overlord/configstate/configcore/picfg.go

```diff
@@ -92,13 +95,53 @@ func updatePiConfig(path string, config map[string]string) error {
 	return nil
 }
 
-var (
-	errPiConfigNotSupported = fmt.Errorf("configuring pi-config not supported in current mode")
-)
+type piConfigNotSupportedError struct {
+	reason string
+}
+
+func newPiConfigNotSupportedError(msg string) *piConfigNotSupportedError {
+	return &piConfigNotSupportedError{msg}
+}
+
+func (e *piConfigNotSupportedError) Error() string {
+	return fmt.Sprintf("configuring not supported: %s", e.reason)
+}
+
+var reIgnorePrefix = regexp.MustCompile(`(?i)^#\s+Snapd-Edit:\s+no\s*$`)
+
+func piConfigFileIgnoreMarkerSet(configFile string) bool {
+	f, err := os.Open(configFile)
+	if err != nil {
+		return false
+	}
+	defer f.Close()
+
+	scanner := bufio.NewScanner(f)
+	// read the first line
+	scanner.Scan()
+	if scanner.Err() != nil {
+		return false
+	}
+
+	return reIgnorePrefix.MatchString(scanner.Text())
```

Comment:
probably no need to convert to strings
```suggestion
	return reIgnorePrefix.Match(scanner.Bytes())
```